autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectColumn.tsx

```diff
@@ -22,12 +25,34 @@ const CourseSelectColumn: React.FC = () => {
   );
   const term = useSelector<RootState, string>((state) => state.termData.term);
   const dispatch = useDispatch();
+  const [loading, setLoading] = React.useState(true);
+  const [wasCourseRemoved, setCourseRemoved] = React.useState(false);
+
+  const removeCallback = React.useCallback((id: number) => {
+    dispatch(removeCourseCard(id));
+    setCourseRemoved(true);
+  }, [dispatch, setCourseRemoved]);
+
+  const resetAnimations = React.useCallback(() => {
+    setCourseRemoved(false);
+  }, []);
 
   const expandedRowRef = React.useRef<HTMLDivElement>(null);
   // Use dynamic className to style expanded card
   React.useLayoutEffect(() => {
     if (expandedRowRef.current) {
-      const expandedRowHeight = expandedRowRef.current.children[0].clientHeight;
+      // calculate the height of the expanded card
+      const cardEl = expandedRowRef.current.getElementsByClassName(cardStyles.card)[0];
+      const header = cardEl.getElementsByClassName(cardStyles.header)[0] as HTMLElement;
+      const content = cardEl.getElementsByClassName(cardStyles.content)[0] as HTMLElement;
+      const sectionRows = cardEl.getElementsByClassName(sectionStyles.sectionRows);
+      // the actual height of content should include the fully expanded section rows, if present
+      const deltaRowHeight = sectionRows.length > 0
+        ? sectionRows[0].scrollHeight - sectionRows[0].clientHeight
+        : 0;
+      const expandedRowHeight = header.scrollHeight + content.scrollHeight
+        + parseFloat(getComputedStyle(content).marginTop) + deltaRowHeight;
+
```

Comment:
Do you think we could turn this into a function? It looks really funky and I think adding a function with a good name + description providing context would help a lot